scripts/ci/reborn_changed_coverage.py

```
@@ -1,204 +1,570 @@
+def mechanically_uninstrumentable_line(source_line: str) -> bool:
+    """Return whether a Rust source line is syntactic scaffolding, not behavior."""
+    stripped = source_line.strip()
     return (
-        PRODUCT_SOURCE_RE.fullmatch(path) is not None
-        and "/tests/" not in path
-        and not path.endswith("/tests.rs")
+        not stripped
+        or stripped.startswith("//")
+        or stripped.startswith("#[")
+        or re.match(r"^(?:pub(?:\([^)]*\))?\s+)?use\b", stripped) is not None
+        or re.fullmatch(r"[{}][,;]?", stripped) is not None
```

Ruff (0.16.0)

[warning] Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)